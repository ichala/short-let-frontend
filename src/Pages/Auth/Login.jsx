import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';
import UserLogin from '../../Api/ApiCalls';

function Login() {
  const [FormData, setFormData] = useState({});
  const [error, setError] = useState(null);
  const [Loading, setLoading] = useState(false);

  const onChange = (e) => {
    setFormData((prevState) => ({
      ...prevState,
      [e.target.name]: e.target.value,
    }));
  };

  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);
  if (user) {
    return <Navigate to="/" replace />;
  }
  const handleSubmit = (e) => {
    e.preventDefault();
    UserLogin(FormData, dispatch, setError, setLoading);
  };
  return (
    <>

      <div className="heading">
        <h1>Login</h1>
        <hr />
      </div>
      {error && (
      <div className="alert alert-danger" role="alert">
        {error}
      </div>
      )}
      <div className="row">
        <div className="col-md-4">
          <div className="alert alert-primary text-sm" role="alert">
            <b className="m-2">User : </b>
            user@user.com:user123
            <br />
            <b className="m-2">Admin : </b>
            admin@admin.com:admin123
          </div>
        </div>
      </div>

      <form onSubmit={(e) => handleSubmit(e)}>
        <div className="mb-3">
          <div id="emailHelp" className="form-text">Email</div>
          <input type="email" required onChange={onChange} name="email" className="form-control" />
        </div>
        <div className="mb-3">
          <div id="emailHelp" className="form-text">Password</div>
          <input onChange={onChange} required name="password" type="password" className="form-control" />
        </div>

        {Loading ? (<div className="spinner-grow text-sm text-center text-success" role="status" />) : (<button type="submit" className="btn btn-success">Login</button>)}

      </form>
    </>
  );
}

export default Login;
