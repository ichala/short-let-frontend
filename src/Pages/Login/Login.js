import { useRef, userState, useEffect} from 'react';

const login = () => {
	const userRef = userRef():
	const userRef = userRef():

	const [user, setUser] = userState('');
	const [pwd, setPwd] = userState('');
	const [errMsg, setErrMsg] = userState('');
	const [success, setSuccess] = userState('');

	userEffect(() => {
		useRef.current.focus();
	}, [])

	userEffect(() => {
		setErrMsg('');
	}, [user, pwd])

	const handlesubmit = async (e) => {
		e.preventDefault();
        console.log(user,pwd);
		setUser('');
		setPwd('');
		setSuccess(true);
	}

	return (
		<>
		  success ? (
		  	<section>
		  	  <h1> You are Logged in </h1>
		  	 <br/>
		  	 <p>
		  	    <a href="#">Go to Home<a/>
		  	 </p>
		  	 </section>
		  	) : (
		<section>
            <p ref={errRef} className={errMsg ? "errmsg" : 
            "offscreen"} aria-live="assertive">{errMsg}</p>
            <h1> Sign In</h1>
            <form onSubmit=(handlesubmit)> 
            <label htmlFor="username">Username:</label>
            <input 
                type="text"
                id="username"
                ref={useRef}
                autoComplete="off"
                onChange={(e)=>setuser(e.target.value)}
                value={user}
                required
                /> 

            <label htmlFor="password">Password:</label>
            <input 
                type="password"
                id="password"
                onChange={(e)=>setpwd(e.target.value)}
                value={user}
                required
                /> 

                <button>Sign In</button> 
           </form>
           <p> 
               need an Account? <br/>
               <span className="line">
           {/* put router link here*/}
           <a href="#">sign Up</a>
		</section>
		)}
		</>
    )
}

export default login