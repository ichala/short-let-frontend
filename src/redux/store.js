import { configureStore } from '@reduxjs/toolkit';
import authSlices from './slices/auth';

export default configureStore({
  reducer: {
    user: authSlices,
  },
});
