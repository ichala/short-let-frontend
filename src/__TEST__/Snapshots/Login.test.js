import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/store';
import Login from '../../Pages/Auth/Login';

describe('Login', () => {
  it('should render the login page', () => {
    const app = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });
});
