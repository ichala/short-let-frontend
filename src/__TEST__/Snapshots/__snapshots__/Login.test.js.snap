// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="heading"
      >
        <h1>
          Login
        </h1>
        <hr />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="alert alert-primary text-sm"
            role="alert"
          >
            <b
              class="m-2"
            >
              User : 
            </b>
            user@user.com:user123
            <br />
            <b
              class="m-2"
            >
              Admin : 
            </b>
            admin@admin.com:admin123
          </div>
        </div>
      </div>
      <form>
        <div
          class="mb-3"
        >
          <div
            class="form-text"
            id="emailHelp"
          >
            Email
          </div>
          <input
            class="form-control"
            name="email"
            required=""
            type="email"
          />
        </div>
        <div
          class="mb-3"
        >
          <div
            class="form-text"
            id="emailHelp"
          >
            Password
          </div>
          <input
            class="form-control"
            name="password"
            required=""
            type="password"
          />
        </div>
        <button
          class="btn btn-success"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      class="heading"
    >
      <h1>
        Login
      </h1>
      <hr />
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-4"
      >
        <div
          class="alert alert-primary text-sm"
          role="alert"
        >
          <b
            class="m-2"
          >
            User : 
          </b>
          user@user.com:user123
          <br />
          <b
            class="m-2"
          >
            Admin : 
          </b>
          admin@admin.com:admin123
        </div>
      </div>
    </div>
    <form>
      <div
        class="mb-3"
      >
        <div
          class="form-text"
          id="emailHelp"
        >
          Email
        </div>
        <input
          class="form-control"
          name="email"
          required=""
          type="email"
        />
      </div>
      <div
        class="mb-3"
      >
        <div
          class="form-text"
          id="emailHelp"
        >
          Password
        </div>
        <input
          class="form-control"
          name="password"
          required=""
          type="password"
        />
      </div>
      <button
        class="btn btn-success"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
